pkg/oauth/registry/oauthaccesstoken/etcd/etcd.go

```diff
@@ -0,0 +1,55 @@
+package etcd
+
+import (
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	kerrs "github.com/GoogleCloudPlatform/kubernetes/pkg/api/errors"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/generic"
+	etcdgeneric "github.com/GoogleCloudPlatform/kubernetes/pkg/registry/generic/etcd"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/tools"
+
+	"github.com/openshift/origin/pkg/oauth/api"
+	"github.com/openshift/origin/pkg/oauth/registry/oauthaccesstoken"
+)
+
+// rest implements a RESTStorage for access tokens against etcd
+type REST struct {
+	etcdgeneric.Etcd
+}
+
+const EtcdPrefix = "/registry/oauth/accesstokens"
+
+// NewREST returns a RESTStorage object that will work against access tokens
+func NewREST(h tools.EtcdHelper) *REST {
+	store := &etcdgeneric.Etcd{
+		NewFunc:     func() runtime.Object { return &api.OAuthAccessToken{} },
+		NewListFunc: func() runtime.Object { return &api.OAuthAccessTokenList{} },
+		KeyRootFunc: func(ctx kapi.Context) string {
+			// TODO: JTL: switch to NoNamespaceKeyRootFunc after rebase
```

Comment:
sadly, no... it changes the method signature - https://github.com/GoogleCloudPlatform/kubernetes/pull/6201